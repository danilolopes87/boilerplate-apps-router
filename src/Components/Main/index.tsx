import * as S from './style'
import React from 'react';

const Main: React.FC = () => {
  return (
    <S.Wrapper>
      <h1>React AvanÃ§ado</h1>
    </S.Wrapper>
  );
};

export default Main;
