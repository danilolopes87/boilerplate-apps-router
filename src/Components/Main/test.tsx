import { render, screen } from '@testing-library/react';
import Main from './index'; // Ajuste o caminho conforme necessário
import React from 'react';

describe('<Main />', () => {
  it('should render heading', () => {
    //renderiza o componente
    const {container} = render(<Main />);

    //busca o elemento e verifica a existência dele
    expect(
      screen.getByRole('heading', { name: /react avançado/i})
    ).toBeInTheDocument()

    //gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  });
});
