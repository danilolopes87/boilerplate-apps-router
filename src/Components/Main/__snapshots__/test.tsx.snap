// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
<main
  class="sc-blHHSb ePrLHj"
>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
